app/views/teachers/TeacherStudentView.coffee

```diff
@@ -132,9 +132,9 @@ module.exports = class TeacherStudentView extends RootView
     @levelStudentCodeMap = {}
     for session in @sessions.models when session.get('creator') is @studentID
       # Normal level
-      @levelStudentCodeMap[session.get('level').original] = session.get('code')?['hero-placeholder']?['plan']
+      @levelStudentCodeMap[session.get('level').original] = [session.get('code')?['hero-placeholder']?['plan']]
       # Arena level
-      @levelStudentCodeMap[session.get('level').original] ?= session.get('code')?['hero-placeholder-1']?['plan']
+      @levelStudentCodeMap[session.get('level').original].push(session.get('code')?['hero-placeholder-1']?['plan'])
```

Comment:
It's a little bit messy to be pushing `undefined` elements into the array and filtering them out later; any other places we use this code map will have to also filter them out or break, and we won't notice it until we hit the relatively rare case of a student who has played on both sides of an arena. I would convert this to if-statement and only push the extra element if it exists.

Reply:
okay, i will update it